modules/centrifuge_classification.nf

```diff
@@ -0,0 +1,103 @@
+// This file contains workflow to classify with centrifuge
+include { unpack_taxonomy } from '../modules/kraken_server'
+
+process unpack_database {
+    label "process_single"
+    storeDir "${params.store_dir}/centrifuge"
+    input:
+        val database
+    output:
+        path "database_dir", emit: database
+    """
+    mkdir database_dir
+    cd database_dir
+    wget "${database}"
+    tar xzf *tar.gz
+    cd ..
+    """
+}
+
+process centrifuge {
+
+    label "process_higher_memory"
+
+    container "docker.io/ontresearch/centrifuge:latest"
+    publishDir "${params.outdir}/${unique_id}/classifications", mode: 'copy'
+
+    input:
+        tuple val(unique_id), path(fastq)
+        path database
+    output:
+        tuple val(unique_id), path("centrifuge_assignments.tsv"), emit: assignments
+        tuple val(unique_id), path("centrifuge_summary.tsv"), emit: summary
+    script:
+    """
+    centrifuge -x "${database}/${params.centrifuge_db_name}" -U ${fastq} \
+        -S centrifuge_assignments.tsv \
+        --report-file centrifuge_summary.tsv
+    """
+}
+
+process centrifuge_report {
+
+    label 'process_low'
+
+    container "docker.io/ontresearch/centrifuge:latest"
```

Comment:
how does this interact with quay.io being set as the docker repository by `-profile docker`?

Reply:
This handles it. Without the `docker.io` it would look on `quay.io`, but with it it looks on dockerhub

Reply:
ah very good